### DISCUSSION TITLE
Qooxdoo Wish List

### DISCUSSION THREAD

**Discussion by goldim** [qooxdoo/qooxdoo#10276] [Ideas]
There are what I would like to see and have further in the next versions of _qooxdoo_. Maybe some functionality or features already implemented and I just don't know about them. Let me know in discussion. So what would be great in my opinion:

1. way to create and use **es6** classes in _qooxdoo_ project
2. inheritance from _qooxdoo_ classes something like that `class MyClass extends qx.core.Object`
3. npm packages instead of qooxdoo packages. Instead of qooxdoo repository to use global npm repository.
4. No need to setup unit tests (installing dependencies, edit config files etc). Just run command which executes tests in console.

**Comment by johnspackman** [comment]
Hi Dmitry

Thanks for the list 👍 

(1) and (2) are kind of on the radar already, although I think that the last time we talked about it there are features that Qooxdoo supports which ES6 does not, mostly to do with properties; that wouldn't prevent the ES6 from extending Qooxdoo classes - assuming that we can find a way to extend ES6 classes from prototypical classes, or that we can convert the implementation of Qooxdoo classes to ES6 native classes.

A related feature which is on my personal wish list is for properties to be accessible via getters and setters (eg `myObject.someValue` instead of `myObject.getSomeValue()`) and also for arrays which are backed by `qx.data.Array` to use `[]` instead of `.getItem()`

(3) Is a good point, AFAICR we considered that and decided to use our own package system instead; I can't remember why exactly, but I think it is to do with having finer control over the packages.  Are there specific benefits/disadvantages you think we would see from using `npm` as opposed to our own?

**Reply by goldim** [reply]
About (3)

- user-friendly for new developers which will discover _qooxdoo_ and start working with it. They already know npm package structure for example
- most of libraries, JS frameworks and projects are distributed via npm official repository
- I would like to make my own private respository and deploy on it my qooxdoo packages. After setup my local projects work with it. AFAIK I can't do that or don't know how to do that.
- if talking about  _qooxdoo_ projects it is more reliable to have them in popular stable repository and the qooxdoo team won't care about some troubles like server shutdowns.

**Reply by cboulanger** [reply]
We write about the reasons very briefly here: https://qooxdoo.org/documentation/6.0/#/development/cli/packages?id=qx-package-and-npm

One reason was that releasing libraries via GitHub already provides everything needed for a package system, npm would not add any value to it (other than, as you say, being already well known), and just add a redundant layer (to have to keep a `package.json` file on top of `Manifest.json. 

I think your use cases can be solved within the current system. I suggest you open a feature request which details what you want to do and how the current system is unhelpful in that respect.
